src/Repository/Doctrine/TransactionRepository.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Adshares\AdsOperator\Repository\Doctrine;
+
+use Adshares\Ads\Entity\Transaction\AbstractTransaction;
+use Adshares\AdsOperator\Repository\TransactionRepositoryInterface;
+
+class TransactionRepository extends BaseRepository implements TransactionRepositoryInterface
+{
+    /**
+     * @return array
+     */
+    public function availableSortingFields(): array
+    {
+        return [
+            'id',
+            'blockId',
+            'type',
+        ];
+    }
+
+    /**
+     * @param string $transactionId
+     * @return AbstractTransaction|null
+     * @throws \Doctrine\ODM\MongoDB\LockException
+     * @throws \Doctrine\ODM\MongoDB\Mapping\MappingException
+     */
+    public function getTransaction(string $transactionId):? AbstractTransaction
+    {
+        /** @var AbstractTransaction $transaction */
+        $transaction = $this->find($transactionId);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$transaction". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)